### PR TITLE
Move MD files to React

### PR DESCRIPTION
**Pull request by AkalUstat**
changelog, shortcut legend, and help moved to react. deleted all unnecessary files

Changes: 
- instead of an html file for each md file, there is one generic html file, and all data is passed to it via IPC

other fixes:

to render react, we were using `app()`. This is causes a problem with hooks, and following the react docs, it has been changed to `React.createElement(app)`.

### CODE REVIEW

**Comment by AkalUstat** [comment]
Please dont merge this yet, shortcut legend markdown is broken

**Comment by AkalUstat** [comment]
ok fixed, this is ready for review

File: app.js [hunk 1]
```diff
@@ -73,7 +73,8 @@ let endChangelogOpenTimer;
 const secondaryWindows = {
   changelogWindow: {
     obj: false,
-    url: `file://${__dirname}/www/changelog.html`,
+    name: 'changelog',
+    title: 'Changelog',
```
Comment (tsingh777) [inline app.js:77]: Titles should be translated and come from `i18n.t`

File: q [hunk 2]
```diff
@@ -0,0 +1,11 @@
+add title to html
```
Comment (tsingh777) [inline q:1]: ? remove

File: www/index.html [hunk 3]
```diff
@@ -70,9 +70,10 @@
       <div id="navigator" class="noselect"></div>
     </div>
     <script>
+      const React = require('react');
       const app = require('./js/app.js').default;
       const { render } = require('react-dom');
-      render(app(), document.getElementById('navigator'));
+      render(React.createElement(app), document.getElementById('navigator'));
```
Comment (tsingh777) [inline www/index.html:76]: can you elaborate on what problem this is solving? Our babel transpilation should handle all this.

File: www/main/addons/secondary-window/Markdown.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
+import ReactMarkdown from 'react-markdown';
+import { ipcRenderer } from 'electron';
+import marked from 'marked';
+import fs from 'fs';
+import path from 'path';
+
+const MarkdownFiles = {
+  help: '../../../../HELP.md',
```
Comment (tsingh777) [inline www/main/addons/secondary-window/Markdown.jsx:9]: we should move these somewhere else. they don't need to be in the root folder. @inderpreetsingh please advice where we should move this.

File: www/main/addons/secondary-window/Markdown.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
+import ReactMarkdown from 'react-markdown';
```
Comment (tsingh777) [inline www/main/addons/secondary-window/Markdown.jsx:2]: Do we need both `ReactMarkdown` and `marked`?

**Comment by tsingh777** [comment]
Collapsible markdown wasn't working when I ran this locally.

File: www/index.html (hunk 3 above)
Comment (AkalUstat) [reply www/index.html:76]: for some reason its not...it errors out esp in secondary-window.html with hooks. Im not sure if babel is messing up but something is going wrong

File: www/main/addons/secondary-window/Markdown.jsx (hunk 5 above)
Comment (AkalUstat) [reply www/main/addons/secondary-window/Markdown.jsx:2]: marked is to properly read the file (tables dont get parsed properly). React markdown is to render the markdown

**Comment by AkalUstat** [comment]
> Collapsible markdown wasn't working when I ran this locally.

yes thats the one thing i couldnt get to work

File: q (hunk 2 above)
Comment (AkalUstat) [reply q:1]: oops wrong vim command 😅

File: www/index.html (hunk 3 above)
Comment (AkalUstat) [reply www/index.html:76]: to add if i did things like `<Markdown />` it wouldnt parse jsx

**Comment by maneetpaul** [comment]
Hey @AkalUstat any update on this one?

**Comment by AkalUstat** [comment]
was this completed with any other pr? if not then i can probably finish it up this week

**Comment by Gauravjeetsingh** [comment]
Closing this for now, will revive when we work again on MD files.

**Pull request closed without merging**